### ISSUE TITLE
train with speed dropout

### ISSUE THREAD

**Issue by kunibald413** [100oylz/IndexTTS2#2]
https://github.com/100oylz/IndexTTS2/blob/0f39865a29482c84db48fe510dcab0fce0868364/indextts/gpt/model_v2.py#L585

try training with papers speed dropout

```python
            # https://arxiv.org/pdf/2506.21619 30% free-form training,
            use_speed = (torch.rand(1) < 0.7).to(device)
```

and build conds like this:

```python
    use_speed = use_speed.long()
    duration_emb = model.mel_pos_embedding.emb(code_lengths) * use_speed.unsqueeze(1)
    duration_emb_half = model.mel_pos_embedding.emb(code_lengths // 2) * use_speed.unsqueeze(1)
    speed_emb = model.speed_emb(torch.zeros_like(use_speed)) * (1 - use_speed).unsqueeze(1)
    duration_emb = duration_emb + speed_emb
    speed_emb_half = model.speed_emb(torch.ones_like(use_speed)) * (1 - use_speed).unsqueeze(1)
    duration_emb_half = duration_emb_half + speed_emb_half

    conds = torch.cat([
        speech_conditioning_latent + emo_vec.unsqueeze(1),
        duration_emb_half.unsqueeze(1),
        duration_emb.unsqueeze(1)
    ], dim=1)
```
